src/amo/components/Addon/index.js

```diff
@@ -546,10 +553,13 @@ export function mapStateToProps(state, ownProps) {
   const { slug } = ownProps.params;
   let addon = getAddonBySlug(state, slug);
 
+  // make sure slug is number or string without leading -
   // It is possible to load an add-on by its ID but in the routing parameters,
   // the parameter is always named `slug`.
   // eslint-disable-next-line no-restricted-globals
-  if (slug && !isNaN(slug)) {
+  if (slug && !isNaN(slug)
+    && ((typeof slug === 'string' && slug.charAt(0) !== '-')
+    || (typeof slug === 'number'))) {
```

Comment:
This makes me think that we should probably extract all these checks into a single utility function above the component.